api/files/views.py

```diff
@@ -35,7 +35,249 @@ def get_file(self, check_permissions=True):
+    it.  If a file is not checked out, this url will be null.
+
+    ###Files (*folders*)
+
+    The `files` endpoint lists all of the subfiles and folders of the current folder. Will be null for files.
+
+    ###Versions (*files*)
+
+    The `versions` endpoint provides version history for files.  Will be null for folders.
+
+    ##Links
+
+        info:        the canonical api endpoint for the latest version of the file
+        new_folder:  url to target when creating new subfolders (null for files)
+        move:        url to target for move, copy, and rename actions
+        upload:      url to target for uploading new files and updating existing files
+        download:    url to request a download of the latest version of the file (null for folders)
+        delete:      url to target for deleting files and folders
+
+    ##Actions
+
+    The `links` property of the response provides endpoints for common file operations. The currently-supported actions
+    are:
+
+    ###Get Info (*files, folders*)
+
+        Method:   GET
+        URL:      links.info
+        Params:   <none>
+        Success:  200 OK + file representation
+
+    The details of a particular file can be retrieved by performing a GET request against the `info` link. The response
+    will be a standard OSF response format with the [OSF File attributes](#attributes).
+
+    ###Download (*files*)
+
+        Method:   GET
+        URL:      links.download
+        Params:   <none>
+        Success:  200 OK + file body
+
+    To download a file, issue a GET request against the `download` link.  The response will have the Content-Disposition
+    header set, which will will trigger a download in a browser.
+
+    ###Create Subfolder (*folders*)
+
+        Method:       PUT
+        URL:          links.new_folder
+        Query Params: ?name={new_folder_name}
+        Body:         <empty>
+        Success:      201 Created + new folder representation
+
+    You can create a subfolder of an existing folder by issuing a PUT request against the `new_folder` link.  The name
+    of the new subfolder should be provided in the `name` query parameter.  The response will contain a [WaterButler
+    folder entity](#folder-entity).
+
+    ###Upload New File (*folders*)
+
+        Method:       PUT
+        URL:          links.upload
+        Query Params: ?kind=file&name={new_file_name}
+        Body (Raw):   <file data (not form-encoded)>
+        Success:      201 Created + new file representation
+
+    To upload a file to a folder, issue a PUT request to the folder's `upload` link with the raw file data in the
+    request body, and the `kind` and `name` query parameters set to `'file'` and the desired name of the file.  The
+    response will contain a [WaterButler file entity](#file-entity) that describes the new file.
+
+    ###Update Existing File (*file*)
+
+        Method:       PUT
+        URL:          links.upload
+        Query Params: ?kind=file&name={new_file_name}
```

Comment:
Should name be `{old_file_name}`?
You can also put directly to a file to update it.